libcorn/src/lua.rs

```diff
@@ -0,0 +1,51 @@
+use mlua::prelude::*;
+use crate::Value;
+use mlua::Value as OtherValue;
+impl<'lua> ToLua<'lua> for Value<'lua> {
+    fn to_lua(self, lua: &'lua Lua) -> LuaResult<LuaValue<'lua>> {
+        match self {
+            Value::Object(map) => {
+                let table = lua.create_table()?;
+                for (k, v) in map {
+                    table.set(k, v)?;
+                }
+                Ok(mlua::Value::Table(table))
+            }
+            Value::Array(vec) => {
+                let table = lua.create_table()?;
+                for (i, v) in vec.into_iter().enumerate() {
+                    table.set(i + 1, v)?;
+                }
+                Ok(mlua::Value::Table(table))
+            }
+            Value::String(s) => Ok(s.to_lua(lua)?),
+            Value::Integer(i) => Ok(i.to_lua(lua)?),
+            Value::Float(f) => Ok(f.to_lua(lua)?),
+            Value::Boolean(b) => Ok(b.to_lua(lua)?),
+            Value::Null(_) => Ok(LuaValue::Nil),
+        }
+    }
+}
+
+fn lua_parse(lua: &Lua, config: String) -> LuaResult<LuaTable> {
+    let globals = lua.globals();
+    let res = crate::parse(&config);
+    let table = lua.create_table()?;
+    match res {
+        Ok(v) => {
+            let val = mlua::ToLua::to_lua(v, lua)?;
+            table.set("value", val)?;
+        }
+        Err(e) => {
+            table.set("error", e.to_string())?;
+        }
+    }
+    Ok(table)
+}
+
+#[mlua::lua_module]
+fn libcorn(lua: &Lua) -> LuaResult<LuaTable> {
+    let exports = lua.create_table()?;
+    exports.set("parse", lua.create_function(lua_parse)?)?;
```

Comment:
These are better separated imo, because it separates the error handling, and makes it a little easier to catch which line fails if they do for any reason:

```suggestion
    let function = lua.create_function(lua_parse)?;
    exports.set("parse", function)?;
```